### ISSUE TITLE
Why does `ContractEntryDefinition` does not fit what webx-codegen outputs ?

### ISSUE THREAD

**Issue by megaturbo** [xf00f/web3x#55]
When I generate my contract types with `web3x-codegen` I end up with `MyContract.ts` and `MyContractAbi.ts` as intended. 

However, in some of my `MyContractAbi.ts`, the `ContractAbi` constructor should take an array of `ContractEntryDefinition`. However some of my entries are set with a `gas` parameter, which does not fit the `ContractEntryDefinition` interface. See below:
```typescript
export default new ContractAbi([
  ...
  {
    "name": "myFunction",
    "outputs": [],
    "inputs": [ ],
    ...
    "gas": 175875
  },
  ...
]);
``` 

Hence I got some compiler errors.

What does it depend on to set this gas parameter ? Should we add it to the `ContractEntryDefinition` interface as an optional parameter ?

**Comment by xf00f** [comment]
Interesting. Can I ask what's generating your ABI file? Do you have any idea what this gas value is meant to be used for? Is it an estimate? If you note the specification it has no mention of gas.
https://solidity.readthedocs.io/en/v0.5.3/abi-spec.html#json

**Comment by megaturbo** [comment]
Oh you're right, and now that you say that I realized it's only my vyper contracts ABIs that shows this gas parameter.

All my ABI files are all generated by `truffle`, hence it should use the vyper compiler [which apparently does a gas estimate](https://github.com/ethereum/vyper/blob/master/vyper/compiler.py#L88)

**Comment by xf00f** [comment]
Merged into v4.0.4.
